block/manager.go

```diff
@@ -728,9 +735,18 @@ func (m *Manager) createNextDABatch(startHeight uint64, endHeight uint64) (*type
 			m.logger.Error("Failed to load commit", "height", height)
 			return nil, err
 		}
+
+		//Check if the batch size is too big
+		totalSize = totalSize + block.ToProto().Size() + commit.ToProto().Size()
```

Comment:
totalSize doesn't equal batch.ToProto().Size() which is what eventually is sent to the DA and the gap gets bigger the bigger the blocks num is.

Reply:
why not?
```
type Batch struct {
	StartHeight uint64
	EndHeight   uint64
	Blocks      []*Block
	Commits     []*Commit
}
```
I want to avoid adding the block to the array and than remove it if batch too large
I think it has the same affect

Reply:
> why not?

I believe it's the extra encoding or padding added from having a list of blocks/comments in the batch.

Either way it's not equal. you can compare and see. 

For example with 1000 empty blocks the `totalSize`: 519006 and the batch size is 524006

If someone will write maxBatchSize 2MB and it will be more de facto, it will fail with sending the batch.